src/collection.ts

```diff
@@ -112,7 +112,11 @@ export interface CollectionOptions
   extends BSONSerializeOptions,
     WriteConcernOptions,
     LoggerOptions {
+  /**
+   * @deprecated Use secondaryOk instead
+   */
   slaveOk?: boolean;
+  secondaryOk?: boolean;
```

Comment:
We should see if this flag is ever actually parsed or used anywhere distinct from readPreference (and why? what should it control that readPreference doesn't?), otherwise, if readPreference is sufficient to cover the use case, we can just say "use readPreference" in the deprecation message. @nbbeeken @durran can you check my logic here?

Reply:
From my understanding `secondaryOk` is required still when communicating with a mongos in conjunction with the read preference, but the driver should be doing this automatically and the user isn't required to do anything here. So I would say your logic is correct that we can say "use read preference" in the deprecation message.

Reply:
I've updated the deprecation note to instruct users to use the readPreference option instead.

I also deleted the slaveOk option and its associated logic in the internal CollectionPrivate interface.  The tests are still passing and the slaveOk option doesn't appear to be used anywhere else. I think this is what Daria and Neal were discussing in Slack yesterday.